[13.0][MIG] l10n_es_aeat_sii: Migration to 13.0

ValentinVinagre opened the pull request:
Status: Ready to review

- [X] Adaptación de código a la v.13.
- [X] Adaptación de código para nuevo cálculo de impuestos. 
- [X] Casos de uso.
- [X] Adaptación de Tests.
- [X] Realización de Rebase.
- [X] Squash de commits administrativos.

acysos committed:
l10n_es_aeat_sii: presentación inmediata de impuesto

Commit:
l10n_es_aeat_sii: Carga directa del certificado

acysos committed:
l10n_es_aeat_sii: se añade mapeo de impuestos, traducciones y permisos

l10n_es_aeat_sii: añadidas facturas de bienes de inversión e intracomunitarias. Se traspasa la configuración del WSDL a parametros

fuentes73 committed:
l10n_es_aeat_sii: Mejoras del certificado + otros

* Ocultar la contraseña del certificado
* Plantilla de cuentas solo obligatorio si "activar SII" activado
* Habilitar un certificado por compañía
* Permisos de acceso
* Tener en cuenta las fechas de los certificados

[IMP] l10n_es_aeat_sii: No utilizar los parámetros del sistema y buscar solo certificados activos

RamonGuiuGou committed:
l10n_es_aeat_sii. Add connector

acysos committed:
l10n_es_aeat_sii: error en la búsqueda del certificado + correción descripciones en facturas de proveedor

ismaelcj committed:
l10n_es_aeat_sii, refactoring

fuentes73 committed:
l10n_es_aeat_sii: Envío de facturas rectificativas

omar7r committed:
[MIG] l10n_es_aeat_sii: Migración a la 10.0

ismaelcj committed:
l10n_es_aeat_sii, tests

RamonGuiuGou committed:
l10n_es_aeat_sii: Filtros + agrupación:

* Añadir filtros a las facturas para saber si el SII se ha enviado o no
* Corregir la agrupacion de los impuestos si en varias línias de factura hay el mismo impuestos

fuentes73 committed:
l10n_es_aeat_sii: Trabajos readonly, limitar campos, poder re-encolar y retardo configurable

acysos committed:
l10n_es_aeat_sii: copy fix, send button, method send, new filter and translations

Jortolsa-S73 committed:
función para verificar si la factura debe enviarse o no a través del SII, pensada para ser heredada en otros módulos como l10n_es_dua_sii

Jortolsa-S73 committed:
añadidos impuestos de importación al mapeo del SII, cambio de NIF

oihane committed:
l10n_es_aeat_sii: update tests

ozono committed:
l10n_es_aeat_sii: función para uso condicional de IDOtro/NIF + inclusión de ImporteTotal

De momento se aplica solo a emitidas

RamonGuiuGou committed:
l10n_es_aeat_sii: Añadir transformación multidivisa en los importes && añadir en la difinicion del producto la configuración de la causa exenta y para poderla enviar a la agencia

oihane committed:
l10n_es_aeat_sii: travis and pylint

pedrorgil committed:
l10n_es_aeat_sii: añadido hook para calcular el precio subtotal por línea de factura. De esta forma permitimos que otros módulos puedan heredar e intervenir en el cálculo del precio subtotal por línea, como por ejemplo para incluir el importe de un tasa, segundos descuentos, etc.

Commit:
l10n_es_aeat_sii:

* filtro facturas sii fallidas + clave registro en posición fiscal
* catch job error on invoice

omar7r committed:
l10n_es_aeat_sii: En lo posición fiscal ahora se permiten poner dos claves por defecto, una para ventas y otra para compras. También se corrigen los problemas de redondeo si se aplica redondeo global en la contabilización de facturas. [FIX] PEP8. [UPD] Traducciones.

RamonGuiuGou committed:
[MIG] l10n_es_aeat_sii: Migration to 9.0

pedrobaeza committed:
l10n_es_aeat_sii: General refactoring:

* Onchanges renamed for avoiding conflicts with other modules
* Methods renamed for avoiding conflicts with other modules
* Some method docstrings.
* Use recordset instead of lists.
* Remove incorrect mix of @api.multi and @api.model decorators
* Remove method arguments that are passed when there are other data that are taken from self.
* Place in the proper model some of the methods for using self reference
* Variables renamed according their use.
* POT file removed.
* Reduce cyclomatic complexity of some methods.
* Code style more homogeneous.
* Uniformize file headers.
* Full Spanish translation.
* Add icon.

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_sii: Make version 9 to work

[IMP] l10n_es_aeat_sii: Change tax computation to invoice level

Odoo v9 already knows which tax generates each invoice tax line, so
we get with this:

* Avoid to make roundings that are not accurate.
* If there are manual changes in supplier invoices taxes, now we
  will be able to notify them and prevent not matching errors.
* Less difficult algorithms for getting tax amount totals.

RamonGuiuGou committed:
l10n_es_aeat_sii: Corrección de pequeños problemas

* TipoImpositivo
* Default del refund_type
* Clave de registro facturas proveedor

ozono committed:
l10n_es_aeat_sii: Facturas exentas + desglose emitidas + ImporteTotal + otros

* Ajuste DesgloseFactura/DesgloseTipoOperacion en emitidas

  Evita/adapta que se incluyan DesgloseFactura/DesgloseTipoOperacion (excluyentes).
  Si es obligatorio usa DesgloseTipoOperacion para ciertas operaciones.
  Se hace al final de _get_sii_out_taxes para tocar lo menos posible lo que ya hay.

* Mapeado S_IVA0_E en SFESBE
* Se añade a recibidas clave ImporteTotal.

RamonGuiuGou committed:
l10n_es_aeat_sii. IR a buscar el TipoImpositivo a los impuestos hijos de los intracomunitarios. Para no presentar el 100% del padre.

ozono committed:
l10n_es_aeat_sii: NoExenta solo cuando corresponda y condiciones

pedrobaeza committed:
l10n_es_aeat_sii: Several improvements:

- Cambio del campo "Carpeta" al asistente de obtención de claves.
- Incluir el tipo de rectificación en el asistente de factura rectificativa.
- Renombrados dos campos que faltaban sin prefijo `sii_` para evitar colisiones con otros módulos.
- Solventadas algunas inconsistencias del UI de la factura.
- Campo readonly para related para evitar efectos laterales
- PEP8

JuanjoA committed:
l10n_es_aeat_sii: Segregación de lógica en _get_sii_invoice_dict

pedrobaeza committed:
l10n_es_aeat_sii: Better tax dict treatment + PEPP8

pedrorgil committed:
Cambiado uso directo del campo price_subtotal de invoice.line por el método "_get_sii_line_price_subtotal" definido para el cálculo de este dato.

ozono committed:
l10n_es_aeat_sii: float_round para redondear resultados

- Se usa float-round en el redondeo
- También se redondean los valores en taxes_to
- Se incluye el redondeo de la BaseImponible
- Se aplica también a recibidas

pedrorgil committed:
l10n_es_aeat_sii: Anulación de facturas + estados + otros:

* Servicio de anulación de facturas presentadas al SII
* Nueva forma de control de estado de presentación SII en base a estados.
* Control de visualización de botones.
* Hook para cálculo de fecha de registro contable de facturas recibidas.

Corregido metodo _get_sii_line_price_subtotal que debía devolver el subtotal (multiplicando por la cantidad) ya que se utiliza para la suma de bases imponibles en algunos sitios, y se ha añadido un nuevo método para cálculo de precio unitario efectivo de las líneas que se utiliza en la llamada al método _compute_all de account.tax.

JuanjoA committed:
l10n_es_aeat_sii: Añadido check para activar o no SII según posición fiscal

pedrobaeza committed:
l10n_es_aeat_sii: Don't populate list on list on 'DetalleIva'

omar7r committed:
l10n_es_aeat_sii: Facturas recibidas exentas, causas de exención y de no sujeción en posición fiscal

pedrorgil committed:
l10n_es_aeat_sii: Signo para rectificativas por diferencia

Las facturas rectificativas por diferencias deben ir con importes negativos.

pedrobaeza committed:
l10n_es_aeat_sii: Tests + tax amount in refund

pedrobaeza committed:
l10n_es_aeat_sii: Last improvements over stable version:

* UI on fiscal positions
* Translations
* Variable name
* Disallow foreign currencies invoices

omar7r committed:
l10n_es_aeat_sii: Convertimos las facturas en otra divisa a € para enviar al SII

ismaelcj committed:
l10n_es_aeat_sii, wrong invoice filter

pedrorgil committed:
Corregido error en anulación de facturas de proveedor (#538)

* [FIX] Corregido error en anulación de facturas de proveedor

* [8.0][l10_es_aeat_sii][version] Bump version

ismaelcj committed:
l10n_es_aeat_sii, invoice description

amcor committed:
l10n_es_aeat_sii: Number on cancelled invoices (#544)

Field 'number' is emptied when canceling invoices, so we take 'internal_number' for that case.

omar7r committed:
l10n_es_aeat_sii: Se corrige la asignación de la clave de registro SII por la posición fiscal

pedrorgil committed:
l10n_es_aeat_sii: Eliminación de trabajos de conector al volver a borrador + estados singleton (#545)

pedrobaeza committed:
l10n_es_aeat_sii: Fixes and translation on description feature

[MIG] l10n_es_aeat_sii: Make module not installable

pedrorgil committed:
[FIX+IMP] l10n_es_aeat_sii: SII versión 1.0 + 1º semestre + otros:

- Se ha modificado el texto de algunas claves de registro para adaptarlos a las
  actualmente publicadas por la AEAT
- Se han añadido las claves de registro correspondientes al primer semestre de 2017
  para facturas emitidas y recibidas
- Se han eliminado dos claves de registro eliminadas por la AEAT en la versión 0.7
- Se ha cambiado el valor de la variable SII_VERSION a '1.0'
- Se han modificado en los parámetros del sistema las URL de los WSDL de SII conforme
  a los publicados por la AEAT. Son funcionales tanto en entorno de pruebas como de
  producción. La versión 0.7 es admitida hasta el día 30 de Junio sólo para el entorno
  de pruebas (en producción debe ser la 1.0), a partir del día 1 de Julio la versión
  debe ser la 1.0 para ambos entornos.
- Se ha eliminado el parámetro del sistema 'l10n_es_aeat_sii.version' ya que no tiene
  sentido en configuración. La versión está hardcodeada en account_invoice.py y no
  debe poder ser configurable por el usuario.
- Se ha contemplado un nueva opción para el campo <TipoNoExenta> que la AEAT añadió en
  la versión 0.7, la opción (S3) que debe utilizarse cuando la factura tiene una parte
  sujeta sin inversión de sujeto pasivo y otra sujeta con inversión de sujeto pasivo.
- Se ha añadido control (provisionalmente) en los envíos para evitar que por error un
  usuario envíe una factura del primer semestre al entorno de producción, comprobándolo
  por el período de la factura. El formato que ahora se genera no está adaptado para
  facturas del primer semestre ya que es algo distinto (hay campos que deben tener un
  valor fijo por defecto, como la descripción que debe ser "Registro del primer
  semestre" o la CuotaDeducible que debe ser 0). Auque he realziado pruebas de envío
  contra la de pruebas con clave de registro del primer semestre y no me ha dado error,
  no se si en producción pasará igual o si, aún no dando el error, la factura estará
  mal comunicada (no tiene la clave correcta o los valores por defecto exigidos). Por
  eso de momento (he añadido un TODO) se evita que se envíen si no se está contra el
  entorno de pruebas. Cuando se adapte el formato de envío y dedidamos cómo abordar el
  tema de la comunicación de las facturas del primer semestre (wizard o similar) lo
  podemos quitar o modificar según veamos.
- Mejorada lógica de comprobación para en envío de facturas
- Corrección de domain para atributo invisible de botones de anulación en formularios de facturas

[FIX] l10n_es_aeat_sii: Cálculo descripción SII para factura automático para todo

[IMP] l10n_es_aeat_sii: Control de cambio de fecha de factura de proveedor y revisión de traducciones

[FIX] l10n_es_aeat_sii: Corrección de fecha programada con metodo en conector 'fixed'

La hora a la que se programan los envíos no respetaba el TZ del usuario

ozono committed:
[FIX + MOD] l10n_es_aeat_sii: Signo BaseImponible + refact. signo

omar7r committed:
[FIX] l10n_es_aeat_sii: Se usa commercial_partner_id en lugar de partner_id (#550)

oihane committed:
[IMP] l10n_es_aeat_sii: varios controles del VAT

* Solo se comprueba que tenga VAT en el caso de regimen nacional
* En caso de extracomunitarias pero con código de pais ES solo acepta
  IDType=07
* Mandar 'NO_DISPONIBLE' en caso de que no haya VAT

oihane committed:
[IMP] l10n_es_aeat_sii: texto más claro para seleccionar causa exención

[FIX] l10n_es_aeat_sii: en caso de que no haya descripción no deja guardar

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Better method for SII invoice description

fuentes73 committed:
[IMP] l10n_es_aeat_sii: Controles sobre el VAT (#1)

* Tener en cuenta el caso regimen nacional y partner no Español
* Si el país no esta establecido, tener el cuenta los dígitos del VAT
* Si no tiene ni pais ni vat devolver vacio

amcor committed:
[8.0][FIX] l10n_es_aeat_sii: fix supplier number required (#555)

* UI adjustments
* Field in refund wizard

pedrorgil committed:
[8.0][l10n_es_aeat_sii][IMP] Mejoras en la usabilidad y corrección de errores

Roodin committed:
[FIX+IMP] l10n_es_aeat_sii:

* Corrige errores y pylint
* post_init_hook
* Corregida referencia a conector
* Permisos y cmapo de posicion fiscal erróneo

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Sii simplified invoices through normal invoices

JuanjoA committed:
[8.0] Cambiado hardcodeo nombres de posición fiscal (#565)

* Cambiado hardcodeado de nombres de posición fiscal a configurable por posición fiscal
* Traducciones

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Seguridad + usabilidad

* Filtro de no enviados SII que sólo incluya 2017
* Los trabajos no realizados se eliminan en lugar de pasarlos a estado "Hecho"
  al reenviar, cancelar, etc.
* Permiso de lectura a queue.job para los gestores de la AEAT.
* Regla de seguridad para que los gestores de la AEAT sólo vean los trabajos
  del SII.
* Los trabajos se encolan como admin para evitar errores al validar por gente
  sin permisos del conector. Desafortunadamente, no se puede mantener el
  usuario original en el trabajo.

[IMP] l10n_es_aeat_sii: Share queue.job view for both types of invoices

[FIX] l10n_es_aeat_sii: Add jobs with sudo for avoiding permission errors

[FIX] l10n_es_aeat_sii: Permissions problem

No more need for connector or AEAT permissions

[IMP] l10n_es_aeat_sii: Track content sent + better layout for return

omar7r committed:
[FIX] l10n_es_aeat_sii: Redondeo + signo + IDType (#566)

* Se redondea la base imponible en el caso de las exentas de clientes
* Nunca se cambian a signo negativo los importes si la factira no es rectificativa
* [Información del NIF en exportaciones a Canarias

pedrorgil committed:
[IMP]l10n_es_aeat_sii: Control de respuesta 'AceptadoConErrores'

fuentes73 committed:
[IMP] l10n_es_aeat_sii: Envío de facturas alquiler de inmuebles

amcor committed:
[IMP] l10n_es_aeat_sii: Servicios intra/extra-comunitarios como no sujetos

pedrorgil committed:
[8.0][aeat_sii][FIX] Poner copy=False en invoice_jobs_ids (#585)

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrorgil committed:
[FIX]l10n_es_aeat_sii: Corrige captura de respuesta para estado 'Parcialmente Correcto'

pedrorgil committed:
[FIX] l10n_es_aeat_sii: Fix Spanish VATs starting with N

Envío correcto para el caso de facturas de bienes o ISP emitidas a
clientes extranjeros con sucursal permanente en España.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Don't write several times on the invoice

Writing on one cursor some invoice values and on error in another,
makes the process to hang due to a lock. Deferring the write to the
end and accumulating values prevents this lock.

Closes #594

omar7r committed:
[FIX+REF] l10n_es_aeat_sii: Refactor country code get + french territories

Refactorización de la obtención del código de país y gestión correcta de
los territorios de ultramar franceses.

omar7r committed:
[FIX] l10n_es_aeat_sii: Jobs invoice + sudo in cancellation

* Se corrige la compañía con la que se crean los trabajos del conector
* Se corrige la creación del trabajo de cancelación sii

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Requeue permissions

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Correct mapping using invoices taxes

As grouped taxes are unfolded on invoice tax footer, we need to
map on SII children taxes (only one of them though), not parent ones

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Update SII technical specification link

omar7r committed:
[IMP] l10n_es_aeat_sii: Claves de registro adicionales (#607)

Se añade la posibilidad de indicar las dos caves de registro adicionales.

oca-transbot committed:
OCA Transbot updated translations from Transifex

omar7r committed:
[FIX] l10n_es_aeat_sii: Se corrige el envío de fcaturas con ivas y no sujetos recibidas

oca-transbot committed:
OCA Transbot updated translations from Transifex

[IMP] l10n_es_aeat_sii: Incluir régimen agrario

[FIX] l10n_es_aeat_sii: Don't mix invoice tax with taxes

[MIG] l10n_es_aeat_sii: Mark module as non installable

[FIX] l10n_es_aeat_sii: Mixing tax line and tax

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Remaining migration fixes

pedrorgil committed:
[8.0][IMP] l10n_es_aeat_sii: Mejor forma de calcular la fecha de registro contable para el SII (#621)

Según la conversación mantenida aquí https://groups.google.com/forum/#!searchin/openerp-spain/sii%7Csort:relevance/openerp-spain/ZM2GlTBI5qw/iOF-FF4YBQAJ , la fecha de registro contable para el SII de las facturas recibidas debe ser la fecha en que se registra la factura en el sistema informático, con independencia de la fecha que se establezca al asiento contable de la misma. Hasta ahora por defecto se establece esta fecha como la fecha del momento del envío (esto es correcto) pero se vuelve a obtener la fecha del momento del envío también para las modificaciones y cancelaciones posteriores, y esto es incorrecto ya que es necesario almacenar la fecha en que se comunica la factura por primera vez para no altararla si hay modificaciones o anulaciones posteriores. Se crea en este PR un campo para almacenar la fecha en que se registra la factura correctamente en el SII por primera vez y se utiliza este dato en los siguientes envíos de la factura al SII. Con esto se independiza completamente la fecha de registro en el SII de la fecha del asiento contable y se asegura además que nunca se comunicará fuera de plazo (8 días ahora, 4 días después) independientemente de la fecha contable que le demos al asiento. Los contables sólo deberán tener en cuenta que deben enviar las facturas que quieran deducirse en un mes antes del día 16 del mes siguiente.
He añadido además una migración para establecer el valor del nuevo campo de fecha de registro en el SII a partir del dato 'FechaRegContable' ya comunicado y que se almacena en el último envío al SII de cada factura.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Look for user current company to see if SII is enabled

In partners, the check is only visible if the company has SII enabled. Before this,
the check was done looking to the partner company, but you can leave this field
empty (specially in a multi-company environment), and then the result is not correct.

Looking to the user company gives us the proper result.

pedrorgil committed:
[IMP] l10n_es_aeat_sii: Añadido mapeo para servicios extracomunitarios

misern2 committed:
[10.0][FIX] Actualización server action a la nueva api

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Añadido control de IVA no deducible

[FIX] l10n_es_aeat_sii: Don't replace taxes in SFRSA, but add them

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: tax instead of tax_line

Cherry-picking of v8 commit doesn't serve as is as the structure is
different in >=v9.

[FIX] l10n_es_aeat_sii: Improve exemption causes in products

* Don't default to 'none'. As field is not required, there's no need for this.
* Filter out when looking for product exemption causes those equal to 'none'.

javierjcf committed:
[MIG] l10n_es_aeat_sii: Migration to 11.0

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Send inmediately if there's a previous error

jabibi committed:
[FIX] Assignment to reserved built-in symbol: type

[FIX] l10n_es_aeat_sii: call _connect_sii() using current invoice

misern2 committed:
[FIX] l10n_es_aeat_sii: Fecha Registro SII (#735)

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: remove incorrect migration scripts

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

omar7r committed:
[FIX] l10n_es_aeat_sii: Enviar los importes de impuesto en la moneda de la compañía (#779)

MouTio committed:
[FIX] l10n_es_aeat_sii: Replaced internal_number with move_name in sii_invoice_dict_out (#773)

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Job channel + job unlink

* On migration, the channel used for the SII jobs has not been respected, and thus
  the record rule is not correct.
* When cancelling an invoice, if you don't see the jobs (for example, an invoicing
  user) the job is not removed, and this is dangerous.

[FIX] l10n_es_aeat_sii: Adapt to changes in account_invoice_refund_link

As the module account_invoice_refund_link was overwriting several fields
and now Odoo includes an standard field for referring to the original
invoice, we use that one for the module

[IMP] l10n_es_aeat_sii: Round rectified amounts by substitution

hugosantosred committed:
[FIX] l10n_es_aeat_sii: Redondear CuotaDeducible

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Reorganize connection + better write back on cancel

* Reduce WDSL service connection code for avoiding to repeat conditions on
  each possible call, using a mapping dictionary that can be extended
  through class variable modification in _register_hook of inherited models
* Fine grain several things on the invoice vals write back on confirm invoice.
* Use the same strategy for writing back values on invoice for cancelling invoice.

[FIX] l10n_es_aeat_sii: Unicode doesn't exist in P3

[FIX] l10n_es_aeat_sii: Make cancel invoice multi-record aware

Fixes #830

jabibi committed:
[IMP] l10n_es_aeat_sii: multiple Tax Agencies (multicompany)

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Make tests inheritable + fixed dates

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Constraint not suitable for multi-company

This constraint leads to integrity errors due to missing values in companies that
doesn't have SII enabled. As the required attribute is set in view, this should be removed.

There's still the chance of writing an invoice without this key by code, but putting
a constraint seems excessive as this key is only needed when sending the invoice. If
not present, there will be an SII error that will help you to identify the problem.

oca-travis committed:
[UPD] Update l10n_es_aeat_sii.pot

angelmoya committed:
IMP SII v1.1

jabibi committed:
[IMP] SII v1.1

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Several refinements for v1.1

* Let as historic data the old agencies
* Filtered properly the exempt causes for lines with that tax
* Raise on multiple exempt causes
* Updated link to 1.1 specification

[IMP] l10n_es_aeat_sii: New README style

jabibi committed:
[FIX] l10n_es_aeat_sii: duplicated xml ID

[FIX] l10n_es_aeat_sii: getting exempt cause error

'product.product' object has no attribute 'tax_line_ids'

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Remove refund option by substituion

This refund mode (SII key 'S') is not fully working as is (it requires 3 invoices), and was
causing problems when using it. It was also understandably the default value!

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: round 'ImporteTotal'

`amount_total_company_signed` can contain unrounded amounts, so we need to round that
amount. We use python `round` for avoiding the same problem in this result.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Avoid access error to queue.job records

Putting groups in the view is not enough for preventing Odoo to load the linked one2many model,
resulting in an error for those without permissions in queue.job object. The solution for this
is to isolate that changes in a separate view with a group, and this way the fields are not
fetched anymore if the user doesn't belong to that group.

There's other view group that also has the same group (AEAT responsible), but shouldn't be
isolated, as there's a field there (sii_state), that it's needed for the visibility of
other elements, and there's no problem with them, as they are not x2x fields.

jabibi committed:
[FIX] l10n_es_aeat_sii: always sending to the test environment

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (204 of 204 strings)

Translation: l10n-spain-11.0/l10n-spain-11.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-11-0/l10n-spain-11-0-l10n_es_aeat_sii/es/

[IMP] l10n_es_aeat_sii: Specific refund invoice type

Add support for specific invoice types on refunds (R1, R2 and R3).

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

eantones committed:
[FIX] l10n_es_aeat_sii: Permission read error when sending an invoice to SII

Introduced on commit https://github.com/OCA/l10n-spain/commit/5067c4ef2b572599f7000403113a4a39236714e9

[FIX] l10n_es_aeat_sii: float_round error in BaseImponible

Example on exempt invoices:

- "Invoice Lines" tab:
    * { quantity: 1, price_unit: 82.60, invoice_line_tax_ids: ['IVA Exento Repercutido'] }

- "SII" tab:
    - "General" tab:

        * Error de envio SII: 1100 | Valor o tipo incorrecto del campo: BaseImponible

    - "Technical" tab:

        * Último contenido enviado al SII:
        {
            "PeriodoLiquidacion": {
                "Ejercicio": 2019,
                "Periodo": "01"
            },
            "IDFactura": {
                "IDEmisorFactura": {
                    "NIF": "XXXXXXXX"
                },
                "FechaExpedicionFacturaEmisor": "07-01-2019",
                "NumSerieFacturaEmisor": "XXXXXXX"
            },
            "FacturaExpedida": {
                "TipoDesglose": {
                    "DesgloseTipoOperacion": {
                        "PrestacionServicios": {
                            "Sujeta": {
                                "Exenta": {
                                    "DetalleExenta": [
                                        {
                                            "BaseImponible": 82.60000000000001,
                                            "CausaExencion": "E1"
                                        }
                                    ]
                                }
                            }
                        }
                    }
                },
                "DescripcionOperacion": "/",
                "ImporteTotal": 82.6,
                "TipoFactura": "F2",
                "ClaveRegimenEspecialOTrascendencia": "01"
            }
        }

        * Retorno SII
        {
            'CSV': None,
            'DatosPresentacion': None,
            'Cabecera': {
                'IDVersionSii': '1.1',
                'Titular': {
                    'NombreRazon': 'XXXXXXXXXXXXXXXXXXXX',
                    'NIFRepresentante': None,
                    'NIF': 'XXXXXXXXXXX'
                },
                'TipoComunicacion': 'A0'
            },
            'EstadoEnvio': 'Incorrecto',
            'RespuestaLinea': [
                {
                    'IDFactura': {
                        'IDEmisorFactura': {
                            'NIF': 'XXXXXXXXXXX'
                        },
                        'NumSerieFacturaEmisor': 'XXXXXXXXXXXXX',
                        'NumSerieFacturaEmisorResumenFin': None,
                        'FechaExpedicionFacturaEmisor': '07-01-2019'
                    },
                    'RefExterna': None,
                    'EstadoRegistro': 'Incorrecto',
                    'CodigoErrorRegistro': 1100,
                    'DescripcionErrorRegistro': 'Valor o tipo incorrecto del campo: BaseImponible',
                    'CSV': None,
                    'RegistroDuplicado': None
                }
            ]
        }

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Update documentation to meet v11 requirements

eantones committed:
[FIX] l10n_es_aeat_sii: float_round error in BaseImponible

Extension of https://github.com/OCA/l10n-spain/pull/972/commits/8a650787f14fd40513e9161a0e78bf09b55fd220

En las lineas https://github.com/OCA/l10n-spain/blob/fabc3a39eda6c595045906a3a612a18a6d902f9f/l10n_es_aeat_sii/models/account_invoice.py#L491-L492
y
https://github.com/OCA/l10n-spain/blob/fabc3a39eda6c595045906a3a612a18a6d902f9f/l10n_es_aeat_sii/models/account_invoice.py#L537-L538

la suma (+=) genera un error de coma flotante, para reproducir el error desde la consola Python:

Python 3.5.2 (default, Nov 12 2018, 13:43:14)
[GCC 5.4.0 20160609] on linux
>>> 170.61 + 107.35
277.96000000000004

Example on customer exempt invoices:

- "Invoice Lines" tab:
    * [
        { quantity: 1,  price_unit: 56.31,  invoice_line_tax_ids: ['IVA Exento Repercutido'] },
        { quantity: 58, price_unit: 0.88,   invoice_line_tax_ids: ['IVA Exento Repercutido'] },
        { quantity: 1,  price_unit: 170.61, invoice_line_tax_ids: ['IVA Exento Repercutido'] },
      ]

- "SII" tab:
    - "General" tab:

        * Error de envio SII: 1100 | Valor o tipo incorrecto del campo: BaseImponible

    - "Technical" tab:

        * Último contenido enviado al SII
        {
            "FacturaExpedida": {
                "ClaveRegimenEspecialOTrascendencia": "01",
                "ImporteTotal": 277.96,
                "DescripcionOperacion": "/",
                "TipoDesglose": {
                    "DesgloseTipoOperacion": {
                        "PrestacionServicios": {
                            "Sujeta": {
                                "Exenta": {
                                    "DetalleExenta": [
                                        {
                                            "BaseImponible": 277.96000000000004,
                                            "CausaExencion": "E1"
                                        }
                                    ]
                                }
                            }
                        }
                    }
                },
                "TipoFactura": "F1",
                "Contraparte": {
                    "NIF": "XXXXXXXX",
                    "NombreRazon": "XXXXXXXXXXXXXXXXXXXXXXXX"
                }
            },
            "IDFactura": {
                "NumSerieFacturaEmisor": "22902",
                "FechaExpedicionFacturaEmisor": "17-01-2019",
                "IDEmisorFactura": {
                    "NIF": "XXXXXXXXXX"
                }
            },
            "PeriodoLiquidacion": {
                "Periodo": "01",
                "Ejercicio": 2019
            }
        }

        * Retorno SII

        {
            'CSV': None,
            'DatosPresentacion': None,
            'Cabecera': {
                'IDVersionSii': '1.1',
                'Titular': {
                    'NombreRazon': 'XXXXXXXXXXXX',
                    'NIFRepresentante': None,
                    'NIF': 'XXXXXXXXXXXXX'
                },
                'TipoComunicacion': 'A0'
            },
            'EstadoEnvio': 'Incorrecto',
            'RespuestaLinea': [
                {
                    'IDFactura': {
                        'IDEmisorFactura': {
                            'NIF': 'XXXXXXXXXXXX'
                        },
                        'NumSerieFacturaEmisor': '22902',
                        'NumSerieFacturaEmisorResumenFin': None,
                        'FechaExpedicionFacturaEmisor': '17-01-2019'
                    },
                    'RefExterna': None,
                    'EstadoRegistro': 'Incorrecto',
                    'CodigoErrorRegistro': 1100,
                    'DescripcionErrorRegistro': 'Valor o tipo incorrecto del campo: BaseImponible',
                    'CSV': None,
                    'RegistroDuplicado': None
                }
            ]
        }

oihane committed:
[IMP] l10n_es_aeat_sii: update wsdl links

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Direct dependency to `l10n_es`

As there's no indirect dependency.

misern2 committed:
[MIG] l10n_es_aeat_sii: Migration to 12.0

luc-demeyer committed:
[IMP] l10n_es_aeat_sii: make account_invoice sii_refund_type default inheritable

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Remove first semester of 2017 check

It's not needed anymore as we are far from that date now.

Commit:
[IMP] l10n_es_aeat_sii: No enviar al SII el mismo contenido (#892)

dalonsod committed:
[11.0][FIX] l10n_es_aeat_sii: on send and cancel error management exception object is not properly cast to string

fuentes73 committed:
[FIX] l10n_es_aeat_sii: queue_job wrong eta value

danielduqma committed:
[IMP] l10n_es_aeat_sii: Default refund specific type

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (211 of 211 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/es/

eantones committed:
[FIX] l10n_es_aeat_sii: float_round error in inbound invoices

Extension of https://github.com/OCA/l10n-spain/pull/985/commits/e63a80fde8b953d09752542eece24451017a065d

Error de redondeo en las facturas entrantes

Example on vendor invoice:

- "Invoice Lines" tab:
    * [
        { quantity: 14,  price_unit: 78.48,  invoice_line_tax_ids: ['IVA Soportado no deducible 21%'] },
      ]

- "SII" tab:
    - "General" tab:

        * Error de envio SII: 1100 | Valor o tipo incorrecto del campo: CuotaSoportada

    - "Technical" tab:

        * Último contenido enviado al SII

        {
            "IDFactura": {
                "IDEmisorFactura": {
                    "NIF": "XXXXXXX"
                },
                "NumSerieFacturaEmisor": "XXXXXX",
                "FechaExpedicionFacturaEmisor": "08-08-2019"
            },
            "PeriodoLiquidacion": {
                "Ejercicio": 2019,
                "Periodo": "08"
            },
            "FacturaRecibida": {
                "TipoFactura": "F1",
                "ClaveRegimenEspecialOTrascendencia": "01",
                "DescripcionOperacion": "/",
                "DesgloseFactura": {
                    "DesgloseIVA": {
                        "DetalleIVA": [
                            {
                                "TipoImpositivo": "21.0",
                                "BaseImponible": 1098.72,
                                "CuotaSoportada": 230.73000000000002
                            }
                        ]
                    }
                },
                "Contraparte": {
                    "NombreRazon": "XXXXXXXXXXXXXX",
                    "NIF": "XXXXXXXXXXXXX"
                },
                "FechaRegContable": "08-08-2019",
                "ImporteTotal": 1329.45,
                "CuotaDeducible": 0.0
            }
        }

        * Retorno SII

        {
            'CSV': None,
            'DatosPresentacion': None,
            'Cabecera': {
                'IDVersionSii': '1.1',
                'Titular': {
                    'NombreRazon': 'XXXXXXXXXXXXXXXXX',
                    'NIFRepresentante': None,
                    'NIF': 'XXXXXXXXXXXXX'
                },
                'TipoComunicacion': 'A1'
            },
            'EstadoEnvio': 'Incorrecto',
            'RespuestaLinea': [
                {
                    'IDFactura': {
                        'IDEmisorFactura': {
                            'NIF': 'XXXXXXXXXXX',
                            'IDOtro': None
                        },
                        'NumSerieFacturaEmisor': 'XXXXXXXXXX',
                        'NumSerieFacturaEmisorResumenFin': None,
                        'FechaExpedicionFacturaEmisor': '08-08-2019'
                    },
                    'RefExterna': None,
                    'EstadoRegistro': 'Incorrecto',
                    'CodigoErrorRegistro': 1100,
                    'DescripcionErrorRegistro': 'Valor o tipo incorrecto del campo: CuotaSoportada',
                    'CSV': None,
                    'RegistroDuplicado': None
                }
            ]
        }

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Apply dict rounding at other level

Doing it at taxes dictionary level, there are some values left that are not
rounded, like 'CuotaDeducible'. We apply the rounding over the top dictionary,
also avoiding to duplicate some code.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Send now button in black color

The assigned class avoided to be shown as it appears as white.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Set refund type on direct refund creation/write

When no context is set and a refund is created, the refund type was not set. Example:

* Create a sales order with stockable product.
* Deliver it and invoice the sales order.
* Return the delivery with refund option checked.
* Invoice the refund.
* No refund type set.

With this patch, we assure that the type is set no matter the context.

MiquelRForgeFlow committed:
[UPD] Remove deprecated from_string()

MiquelRForgeFlow committed:
[UPD] Remove deprecated decodestring and encodestring

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Allow new invoice state `in_payment`

That state is not covered by the code.

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Don't depend on tax description

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Include newly created non deductible taxes in mapping

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Make VAT not subject to work

These were not working since https://github.com/odoo/odoo/commit/5b61898a,
so this fixes this.

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/

angelmoya committed:
Update account_invoice.py

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Include in ROADMAP #1171

Closes #1171

JuanjoA committed:
Extract method to obtain if an invoice should be simplified or not

JuanjoA committed:
sii out_refund invoice with R1 by default

JuanjoA committed:
[FIX] l10n_es_aeat_sii: Envío al SII ImporteTotal sin IRPF en facturas de proveedor

- Añadido aeat.sii.map.lines para NotIncludedInTotal
- facturas emitidas irpf

ValentinVinagre committed:
[MIG] l10n_es_aeat_sii: Migration to 13.0

pedrobaeza commented:
¿Lo sacas con el enfoque que te comenté?

ValentinVinagre commented:
> ¿Lo sacas con el enfoque que te comenté?

He realizado unas primeras pruebas para poder migrar la mayoría de las funcionalidades. Me queda ver como realizarlo bien del todo con los impuestos del 0%, pero si tienes alguna idea ya en la cabeza la incluimos.

pedrobaeza commented:
Bueno, mi idea es examinar los `line_ids` y coger bases de los `tax_ids` y cuotas de los `tax_line_ids`.

ValentinVinagre commented:
> Bueno, mi idea es examinar los `line_ids` y coger bases de los `tax_ids` y cuotas de los `tax_line_ids`.

Si la idea es esa, el problema es cuando se recorren las lineas de impuestos en las funciones de "_get_sii_out_taxes" y "_get_sii_in_taxes" unicamente se recorren las de "tax_repartition_line_id" y estoy mirando a ver como cambiar ese bucle de alguna manera que no se tenga que cambiar toda la lógica.

ValentinVinagre commented:
Hola @pedrobaeza ,
en el último commit he añadido el cambio del cálculo de impuestos, en un principio creo que cuadra todo, ahora calcula bien los impuestos de base 0 y de 0% de aplicación.
En un principio estaría listo para revisar. He añadido estos cambios en un commit separado para que se pueda ver la diferencia y por si se quieren seguir realizando cambios (al final realizaré un squash de todo).

rafaelbn commented:
¿Qué os parece renombar a l10n_es_aeat_sii_oca para que salga en el Apps Store de Odoo?

Comentado en https://github.com/OCA/l10n-spain/issues/1042

rafaelbn commented on l10n_es_aeat_sii/__manifest__.py:
Please change URL to https://www.aeodoo.org/

rafaelbn commented on l10n_es_aeat_sii/__manifest__.py:
Maybe the right one is https://github.com/OCA/l10n-spain/

HaraldPanten commented:
> ¿Qué os parece renombar a l10n_es_aeat_sii_oca para que salga en el Apps Store de Odoo?
> 
> Comentado en #1042

Yo lo veo bien, creo que todos los módulos OCA deberían estar indexados ahí y actualmente el SII no tiene presencia en la App Store.

Me surgen 2 dudas:

1- ¿Qué diferencia hay entre los que están en Odoo Apps hoy por hoy y el del repo de la localización? ¿No es el mismo?

2- ¿Qué hacemos con las versiones previas a la v13?

@acysos @pedrobaeza , como PSCs ¿qué creéis que sería lo mejor?

Saludos

ValentinVinagre commented on l10n_es_aeat_sii/__manifest__.py:
Done!

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Fix re-sending method

Previous method was comparing 2 strings for equality, but coming from dictionaries,
the generated strings may be different, as the order is not guaranteed in
<= Python 3.5.

We can use Python dictionary comparison instead for accurated results.

The method name has also been changed for reflecting the real semantics of it.

TT25025

pedrobaeza commented:
Incluye por favor este commit con un problema que acabo de descubrir: 222b09c88298b1da77b2170c43a42f466355d768

ValentinVinagre commented:
> Incluye por favor este commit con un problema que acabo de descubrir: [222b09c](https://github.com/OCA/l10n-spain/commit/222b09c88298b1da77b2170c43a42f466355d768)

Añadido.

pedrobaeza commented:
Como odoo/odoo#55491 ya está fusionado, se pueden volver a poner los IVA no deducibles del 4 y el 10 aquí.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: VAT not subject for goods

Since odoo/odoo#34044, VAT not subject for goods was not included in any SII taxes
map, which provokes that any vendor bill of this type is not correctly sent.

pedrobaeza commented:
Otro commit a incluir de un bug que he detectado en v12: 4d4367e233168d6fc939ae63d9e9e6bf8b5fb78b

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Compute currency taxes amounts with proper date

If there's an accounting date, taxes amounts in company currency are not properly
computed.

The method has been refactored as well for using the modern `_convert` method.

pedrobaeza commented:
Vaya día! Otro bug que he descubierto! Fix a hacer cherry-pick en 81999879707b4802c1bb9e56d82841bad0fa4e1f

ValentinVinagre commented:
Hola @pedrobaeza ,
Se han añadido los impuestos agregados en https://github.com/odoo/odoo/pull/55491
Se ha añadido el bug detectado en el https://github.com/OCA/l10n-spain/commit/4d4367e233168d6fc939ae63d9e9e6bf8b5fb78b
Sobre el commit 8199987 en la v.13 el modelo account.invoice.tax ya no existe y el modo de cálculo es diferente, por lo que se debería de revisar si el mismo error encontrado en la v.12 pasa en la v.13 ¿Me podrías confirmar este punto?

FernandoRomera commented on l10n_es_aeat_sii/models/account_move.py:
Typo:
cadastral code -> cadastrial code

ValentinVinagre commented on l10n_es_aeat_sii/models/account_move.py:
Creo que es correcto como está:
https://www.wordreference.com/es/en/translation.asp?spen=catastral

FernandoRomera commented on l10n_es_aeat_sii/models/account_move.py:
Ya, si correcto puede ser, simplemente era por coherencia con la definición del siguiente campo:

sii_property_cadastrial_code = fields.Char(string="Real property cadastrial code", size=25, copy=False,)

y es más fácil cambiar la descripción del selector, que el nombre de la columna.

ValentinVinagre commented on l10n_es_aeat_sii/models/account_move.py:
@pedrobaeza que opinas?

pedrobaeza commented on l10n_es_aeat_sii/models/account_move.py:
Son campos no demasiado importantes, y aunque es cierto que es incongruente uno con el otro, cambiarlo significa tener que retraducir, así que si podéis aguantar sin que os chirríe mucho, mejor dejarlo como estaba.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Get default invoice type properly

You can have default invoice type expressed in the context key `default_type`, so
we have to check that as well.

eantones committed:
[FIX] l10n_es_aeat_sii: Added rounding for "Régimen agrario" key ImporteCompensacionREAGYP

pedrobaeza commented:
A tener en cuenta #1462

danielduqma committed:
[FIX] l10n_es_aeat_sii: Default refund specific type

Don't assign a fixed refund specific type for avoiding problems in refunds of
simplified invoices

pedrobaeza committed:
[REF] l10n_es_aeat_sii: Refactor _get_sii_in_taxes for less code

This refactoring is made for reducing code and do it even more readable, but
esentially for a later improvement that will come for grouping taxes with same
percentage.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Merge tax dicts for same TipoImpositivo

If not merged, we may rebase the limit of 6 maximum blocks of this type. This can
happen when using several types of VAT taxes (goods/services) in the same invoice,
or not deductible VAT with different analytic accounts.

TT25683

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_sii: Adjust "total without IRPF" algorithm

- Don't process again everything, but do it inline on existing methods
- Make tests resilient

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: black, isort, prettier

pedrobaeza committed:
[MIG+IMP] l10n_es_aeat_sii: Complete migration

- Redo the tax computation for efficiency and reliability
- Add test suite for checking obtained JSON for different tax configurations
- SII description computation reworked for new computed writable
- Added migration scripts

alex-guardado commented on l10n_es_aeat_sii/views/account_move_views.xml:
Creo que aquí hay un caso no contemplado. Cuando intentas escoger una 'SII registration key' siempre te salen las de cliente. Esto se debe a que el dominio está por defecto en ('type' , '=', 'sale'). Al unificarse las dos vistas 'form' en una en la versión 13, creo que se debería contemplar este caso para las Facturas de proveedor y poner el dominio de ('type' , '=', 'purchase') en este caso . Esto tambien sucede en los campos de "sii_registration_key_additional1" y "sii_registration_key_additional2".

alex-guardado commented on l10n_es_aeat_sii/views/account_move_views.xml:
Creo que aquí hay un caso no contemplado. Cuando intentas escoger una 'SII registration key' siempre te salen las de cliente. Esto se debe a que el dominio está por defecto en ('type' , '=', 'sale'). Al unificarse las dos vistas 'form' en una en la versión 13, creo que se debería contemplar este caso para las Facturas de proveedor y poner el dominio de ('type' , '=', 'purchase') en este caso . Esto tambien sucede en los campos de "sii_registration_key" y "sii_registration_key_additional2".

alex-guardado commented on l10n_es_aeat_sii/views/account_move_views.xml:
Creo que aquí hay un caso no contemplado. Cuando intentas escoger una 'SII registration key' siempre te salen las de cliente. Esto se debe a que el dominio está por defecto en ('type' , '=', 'sale'). Al unificarse las dos vistas 'form' en una en la versión 13, creo que se debería contemplar este caso para las Facturas de proveedor y poner el dominio de ('type' , '=', 'purchase') en este caso . Esto tambien sucede en los campos de "sii_registration_key" y "sii_registration_key_additional1".

pedrobaeza commented on l10n_es_aeat_sii/views/account_move_views.xml:
Gracias por notificarlo. Reviso a ver.

HaraldPanten commented:
Creo que te falla el Travis pero por el Geonames. Está listo para probar funcionalmente, ¿correcto?

pedrobaeza commented:
Sí, falló en ese momento por el glitch temporal. Falta por corregir los impuestos compuestos (intracomunitario, exportación, etc) y la cuestión del dominio de la clave de registro, pero todo lo demás ya está.

HaraldPanten reviewed (approved):
¿Pudiste acabar lo que te faltaba?

Por mi parte lo he estado validando en local con un certificado y los principales casos de uso. A bote pronto no he detectado nada raro. También los intracomunitarios y las exportaciones, por eso que comentabas de los compuestos.

Si están revisados los puntos que te faltaban, y por vuestra parte también se ha testeado "sobre el terreno", por mí podemos hacer fast tracking para no atrasar más el tema.

pedrobaeza committed:
[REN] l10n_es_aeat_sii_oca: Rename from l10n_es_aeat_sii

For avoiding collapse on Odoo Apps Store.

pedrobaeza commented:
Bueno, terminada la revisión de todos los tipos de impuestos y han surgido otros ajustes, pero ya todo parece correcto y tenemos una base de tests buenas para si ocurre cualquier regresión, escribir fácilmente un check para ese problema y que no vuelva a pasar.

He renombrado también el addon (con toda su historia para no perder el historial de commits en futuras versiones) debido a #1042 y no haber más movimientos al respecto.

Procedo a fusionar por tanto tras el largo camino de testeo:

/ocabot merge nobump
